src/modm/driver/radio/sx128x.hpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2023, Lasse Alexander Jensen
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SX128X_HPP
+#define MODM_SX128X_HPP
+
+#include "sx128x_definitions.hpp"
+#include "sx128x_transport.hpp"
+
+namespace modm
+{
+
+/**
+ * @tparam	Transport   Transpot Layer for SX128x
+ *
+ * @ingroup modm_driver_sx128x
+ * @author	Lasse Alexander Jensen
+ * @author  Rasmus Kleist Hørlyck Sørensen
+ */
+template< class Transport >
+class Sx128x : public sx128x, public Transport
+{
+public:
+    Sx128x() = default;
+
+    /// Get the transciever status directly.
+    /// @attention this command can be issued at any time as long as it is not the very first command send over the interface
+    modm::ResumableResult<bool>
+    getStatus(Status *status);
+
+    /// Writes a single byte in a data memory space at the specified address
+    modm::ResumableResult<bool>
+    writeRegister(Register reg, uint8_t data);
+
+    /// Writes a block of bytes in a data memory space starting at a specific address
+    modm::ResumableResult<bool>
+    writeRegister(Register reg, uint8_t *data, size_t length);
```

Comment:
`data` should be `const` here and all the other `write*` functions. We could also consider using `std::span` instead of pointer + length arguments in new code.

Reply:
True I was considering if I should have done that but decided not to in respect of what is currently done in most other drivers. So for write this should be std::span<const uint8_t> and for read std::span<uint8_t>?

Reply:
`std::span<const uint8_t>` for `const` and `std::span<uint8_t>` for non-`const`.